backend-api/src/controllers/vehicle.js

```diff
@@ -85,4 +91,53 @@ const remove = async (req, res) => {
     return res.status(200).json({ message: 'Vehicle is successfully deleted' });
 }
 
-module.exports = { get, all, create, update, remove };
\ No newline at end of file
+
+const readCommand = async (req, res) => {
+    const thingName = req.params.vin;
+    return iotData.getThingShadow({ thingName }, (err, data) => {
+        if (err) {
+            return res.status(400).json({ message: err });
+        }
+        return res.status(200).json(JSON.parse(data.payload).state.reported);
+    });
+};
+
+const writeCommand = async (req, res) => {
+    const thingName = req.params.vin;
```

Comment:
@darkwhitenez right now we don't care much about security, but just so you know, this approach allows anyone to have access to all the cars. least you can do is to verify the authed user by jwt token and extract the vin on the backend and not directly by a param

Reply:
I think the requireCarSubscription() middleware covers it (it's configured in the routes/vehicle.js). Mentioned middleware allows this action only to the users who are subscribed to the car

https://github.com/dsd-m2m/vehicle-tracking/pull/44/commits/a1d41e728880cffce7ecdb69ac44e2b34b29d064#diff-59358a9aa65ea6707904f6646b6911a2

Reply:
didn't know that! Cool.